utilities/run_unit_tests.sh

```diff
@@ -1,55 +1,50 @@
-#!/bin/sh
+#!/bin/bash
 
 usage () {
-    echo -e "\tUSAGE: $0 <main|initial> <device|simulator>"
+    echo -e "\tUSAGE: $0 -f <main|initial> -p <device|simulator>"
     exit 1
 }
 
-set -e
-
-if [ $# -lt 2 ]; then
-    usage
-fi
-
-case $1 in
-    main)
-    ;;
-
-    initial)
-    ;;
+validate_name () {
+    if ! [[ "$FIRMWARE_TYPE" =~ ^Main|Initial$ ]]; then
+        echo -e "Incorrect firmware ($FIRMWARE_TYPE) selected for build\n"
+        usage
+    fi
+}
 
-    *)
-    echo "Wrong type selection"
-    usage
-    ;;
-esac
+validate_platform () {
+    if ! [[ "$BUILD_PLATFORM" =~ ^Device|Simulator$ ]]; then
+        echo -e "Incorrect platform ($BUILD_PLATFORM) selected for build\n"
+        usage
+    fi
+}
 
-case $2 in
-    device)
-    ;;
+set -e
 
-    simulator)
-    ;;
+while getopts 'f:p:' flag; do
+  case "${flag}" in
+    f) FIRMWARE_TYPE=${OPTARG^} ;;
+    p) BUILD_PLATFORM=${OPTARG^} ;;
+    *) usage ;;
+  esac
+done
 
-    *)
-    echo "Wrong platform selection"
-    usage
-    ;;
-esac
+validate_name
+validate_platform
 
 echo "Invoking build script"
-echo "./utilities/build.sh "$1" "unit_tests" "$2""
-./utilities/build.sh "$1" "unit_tests" "$2"
+echo "./utilities/build.sh -c -f \"$FIRMWARE_TYPE\" -t \"unit_tests\" -p \"$BUILD_PLATFORM\""
+./utilities/build.sh -c -f "$FIRMWARE_TYPE" -t "unit_tests" -p "$BUILD_PLATFORM"
 
-if [ "$2" == "device" ]; then
+if [ "$BUILD_PLATFORM" == "device" ]; then
```

Comment:
`BUILD_PLATFORM=${OPTARG^}` in line #27 converts "device" into "Device", therefore this comparison fails